docs/contributor/06-module-migration-guide.md

```diff
@@ -0,0 +1,231 @@
+# Module Migration Guideline
+
+This guide provides detailed instructions for how to migrate a Kyma module from existing (old) module metadata, especially channel-based ModuleTemplates, to the new module metadata, i.e. version-based ModuleTemplates accompanied by ModuleReleaseMeta.
+
+It is highly recommended that module teams familiarize themselves with the new module metadata before starting the migration:
+- The ADR backing the migration is [#984](https://github.com/kyma-project/community/issues/984)
+- An Update on Module Metadata has also been given in the 2024-11-26 Iteration Review
+
+## Target Process
+
+The target process is shown in the figure below. As an example, the `telemetry` module is used. The promotion of module versions and their assignments to channels is managed via three individual processes. In addition, there is a fourth process to delete a module version.
+
+![Target Process](./assets/module-migration.png)
+
+### 1) Submitting a new module version
+
+First, the module developer submits a new module version via a PR to the `/kyma/module-manifests` repository. The submission must provide a `module-config.yaml` file under the path `/modules/<module-name>/<module-version>` where `<module-version>` matches the version of the module configured in `module-config.yaml`. For the detailed format of the `module-config.yaml`, see below.
```

Comment:
The migration guide is directed to module development teams (it is also located in the contributor folder, which already defines the audience). I would suggest making it clear in the intro and then using the second person singular (you) throughout the document instead of "the module developer", etc.